rest/model/project_test.go

```diff
@@ -71,73 +72,109 @@ func TestGitHubDynamicTokenPermissionGroupToService(t *testing.T) {
+		assert.Equal("some-group", model.Name)
+		assert.False(model.AllPermissions)
+		assert.Equal("read", utility.FromStringPtr(model.Permissions.Contents))
+		assert.Equal("write", utility.FromStringPtr(model.Permissions.PullRequests))
 	})
 
 	t.Run("All permissions set should be kept", func(t *testing.T) {
-		req := APIGitHubDynamicTokenPermissionGroups{
+		req :=
 			APIGitHubDynamicTokenPermissionGroup{
 				Name:           utility.ToStringPtr("some-group"),
 				AllPermissions: utility.TruePtr(),
-			},
-		}
-		models, err := req.ToService()
+			}
+		model, err := req.ToService()
 		require.NoError(err)
-		require.Len(models, 1)
+		require.NotNil(model)
 
-		assert.Equal("some-group", models[0].Name)
-		assert.True(models[0].AllPermissions)
+		assert.Equal("some-group", model.Name)
+		assert.True(model.AllPermissions)
 	})
 
 	t.Run("If permissions are set and all permissions is true, an error is returned", func(t *testing.T) {
-		req := APIGitHubDynamicTokenPermissionGroups{
-			APIGitHubDynamicTokenPermissionGroup{
-				Name:           utility.ToStringPtr("some-group"),
-				Permissions:    map[string]string{"contents": "read", "pull_requests": "write"},
-				AllPermissions: utility.TruePtr(),
-			},
+		req := APIGitHubDynamicTokenPermissionGroup{
+			Name:           utility.ToStringPtr("some-group"),
+			Permissions:    map[string]string{"contents": "read", "pull_requests": "write"},
+			AllPermissions: utility.TruePtr(),
 		}
 		_, err := req.ToService()
 		require.ErrorContains(err, "a group will all permissions must have no permissions set")
 	})
 }
+
+func TestGitHubDynamicTokenPermissionGroupBuildFromService(t *testing.T) {
+	require := require.New(t)
+	assert := assert.New(t)
+
+	t.Run("Empty permissions should result in no permissions", func(t *testing.T) {
+		pg := APIGitHubDynamicTokenPermissionGroup{}
+		err := pg.BuildFromService(model.GitHubDynamicTokenPermissionGroup{})
+		require.NoError(err)
+		assert.Equal(utility.FromStringPtr(pg.Name), "")
+		assert.Equal(len(pg.Permissions), 0)
+	})
+
+	t.Run("Valid permissions should be converted", func(t *testing.T) {
+		pg := APIGitHubDynamicTokenPermissionGroup{}
+		err := pg.BuildFromService(model.GitHubDynamicTokenPermissionGroup{
+			Name: "some-group",
+			Permissions: github.InstallationPermissions{
+				Administration: utility.ToStringPtr("write"),
+				Actions:        utility.ToStringPtr("write"),
+				Checks:         utility.ToStringPtr("read"),
+			},
+		})
+		require.NoError(err)
+		require.Equal(len(pg.Permissions), 3)
+		assert.Equal(utility.FromStringPtr(pg.Name), "some-group")
+		assert.Contains(pg.Permissions, "administration")
+		assert.Equal(pg.Permissions["administration"], "write")
+		assert.Contains(pg.Permissions, "actions")
+		assert.Equal(pg.Permissions["actions"], "write")
+		assert.Contains(pg.Permissions, "checks")
+		assert.Equal(pg.Permissions["checks"], "read")
+	})
+
+	t.Run("All permissions set should be kept", func(t *testing.T) {
```

Comment:
Could we edit this to specify that it's the default group for consistency? User's shouldn't be able to make a group that replicates this behavior. I wonder if setting all permissions should error when we build from service since it's not suppose to be possible